worker/src/features/evaluation/evalService.ts

```diff
@@ -240,6 +241,41 @@ export const createEvalJobs = async ({
     }
   }
 
+  const datasetConfigs = configs.filter((c) => c.target_object === "dataset");
+  let cachedDatasetItemIds: { id: string; datasetId: string }[] | null = null;
+  if (datasetConfigs.length > 1) {
```

Comment:
TODO: We should attempt to parallelize the two cache calls. This will increase throughput on the container.